text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,278 @@
+
+* Update default boostrap template to include default stack set roles
+* Create StackSet core construct
+* Create StackSet artifact type in the Cloud Assembly
+* Create StackSet Synthesizer
+* Update CLI to collect StackSet artifact types in addition to Stacks
+* Add deployment code for creating/updating stack sets
+
+### Are there any open issues that need to be addressed later?
+
+* CDK asset handling
+* Future support for service managed stack sets
+* Future support for delete stack set
+* CDK Pipelines
+
+## Appendix
+
+### CDK Bootstrap Template Addition
+
+To support stack sets,
+a stack set administration role should be created in the parent account and a stack set execution role should be created in the child accounts.
+To make this process easier, default roles have been added to the default bootstrap template. (Bootstrapping steps describe in the README section).
+
+```yaml
+  # ...
+  CloudFormationStackSetAdministrationRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Principal:
+              Service: cloudformation.amazonaws.com
+        Version: '2012-10-17'
+      RoleName:
+        Fn::Sub: cdk-${Qualifier}-stack-set-admin-role-${AWS::AccountId}-${AWS::Region}
+  CloudFormationStackSetAdministrationRolePolicy:
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyDocument:
+        Statement:
+          - Action:
+              - sts:AssumeRole
+            Resource:
+              - Fn::Sub: "arn:aws:iam::*:role/cdk-${Qualifier}-stack-set-exec-role-${AWS::Region}"
+            Effect: Allow
+        Version: '2012-10-17'
+      Roles:
+        - Ref: CloudFormationStackSetAdministrationRole
+      PolicyName:
+        Fn::Sub: cdk-${Qualifier}-stack-set-admin-role-policy-${AWS::Region}
+  CloudFormationStackSetExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: sts:AssumeRole
+            Effect: Allow
+            Principal:
+              AWS:
+                Fn::GetAtt: CloudFormationStackSetAdministrationRole.Arn
+          - Fn::If:
+              - HasTrustedAccounts
+              - Action: sts:AssumeRole
+                Effect: Allow
+                Principal:
+                  AWS:
+                    Ref: TrustedAccounts
+              - Ref: AWS::NoValue
+      ManagedPolicyArns:
+        Fn::If:
+          - HasCloudFormationExecutionPolicies
+          - Ref: CloudFormationExecutionPolicies
+          - Fn::If:
+            - HasTrustedAccounts
+            # The CLI will prevent this case from occurring
+            - Ref: AWS::NoValue
+            # The CLI will advertise that we picked this implicitly
+            - - Fn::Sub: "arn:${AWS::Partition}:iam::aws:policy/AdministratorAccess"
+      RoleName:
+        Fn::Sub: cdk-${Qualifier}-stack-set-exec-role-${AWS::Region}
```

Comment:
So, I see that `CloudFormationStackSetAdministrationRole` has `${AWS::AccountId}` in its name, but this one doesn't.

Why the discrepancy?

Reply:
So this is the stack set execution role in the child accounts, which must have the same name across all child accounts. (Adding account id would break that)

Reply:
Ah, of course. This is probably worth a comment in the template body.